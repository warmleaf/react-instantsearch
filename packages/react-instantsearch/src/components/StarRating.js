import React, { Component } from 'react';
import PropTypes from 'prop-types';
import translatable from '../core/translatable';
import classNames from './classNames.js';
const cx = classNames('StarRating');

class StarRating extends Component {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    refine: PropTypes.func.isRequired,
    createURL: PropTypes.func.isRequired,
    min: PropTypes.number,
    max: PropTypes.number,
    currentRefinement: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }),
    count: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.string,
        count: PropTypes.number,
      })
    ),
    canRefine: PropTypes.bool.isRequired,
  };

  static contextTypes = {
    canRefine: PropTypes.func,
  };

  componentWillMount() {
    if (this.context.canRefine) this.context.canRefine(this.props.canRefine);
  }

  componentWillReceiveProps(props) {
    if (this.context.canRefine) this.context.canRefine(props.canRefine);
  }

  onClick(min, max, e) {
    e.preventDefault();
    e.stopPropagation();
    if (
      min === this.props.currentRefinement.min &&
      max === this.props.currentRefinement.max
    ) {
      this.props.refine({ min: this.props.min, max: this.props.max });
    } else {
      this.props.refine({ min, max });
    }
  }

  buildItem({
    max,
    lowerBound,
    count,
    translate,
    createURL,
    isLastSelectableItem,
  }) {
    const disabled = !count;
    const isCurrentMinLower = this.props.currentRefinement.min < lowerBound;
    const selected =
      (isLastSelectableItem && isCurrentMinLower) ||
      (!disabled &&
        lowerBound === this.props.currentRefinement.min &&
        max === this.props.currentRefinement.max);

    const icons = [];
    for (let icon = 0; icon < max; icon++) {
      const iconTheme = icon >= lowerBound ? 'ratingIconEmpty' : 'ratingIcon';
      icons.push(
        <span
          key={icon}
          {...cx(
            iconTheme,
            selected && `${iconTheme}Selected`,
            disabled && `${iconTheme}Disabled`
          )}
        />
      );
    }

    // The last item of the list (the default item), should not
    // be clickable if it is selected.
    const isLastAndSelect = isLastSelectableItem && selected;
    const StarsWrapper = isLastAndSelect ? 'div' : 'a';
    const onClickHandler = isLastAndSelect
      ? {}
      : {
          href: createURL({ min: lowerBound, max }),
          onClick: this.onClick.bind(this, lowerBound, max),
        };

    return (
      <StarsWrapper
        {...cx(
          'ratingLink',
          selected && 'ratingLinkSelected',
          disabled && 'ratingLinkDisabled'
        )}
        disabled={disabled}
        key={lowerBound}
        {...onClickHandler}
      >
        {icons}
        <span
          {...cx(
            'ratingLabel',
            selected && 'ratingLabelSelected',
            disabled && 'ratingLabelDisabled'
          )}
        >
          {translate('ratingLabel')}
        </span>
        <span> </span>
        <span
          {...cx(
            'ratingCount',
            selected && 'ratingCountSelected',
            disabled && 'ratingCountDisabled'
          )}
        >
          {count}
        </span>
      </StarsWrapper>
    );
  }

  render() {
    const { min, max, translate, count, createURL, canRefine } = this.props;

    // min & max are always set when there is a results, otherwise it means
    // that we don't want to render anything since we don't have any values.
    const limitMin = min !== undefined && min >= 0 ? min : 0;
    const limitMax = max !== undefined && max >= 0 ? max : -1;
    const inclusiveLength = limitMax - limitMin + 1;
    const safeInclusiveLength = Math.max(inclusiveLength, 0);

    const values = count
      .map(item => ({ ...item, value: parseFloat(item.value) }))
      .filter(item => item.value >= limitMin && item.value <= limitMax);

    const range = new Array(safeInclusiveLength)
      .fill(null)
      .map((_, index) => {
        const element = values.find(item => item.value === limitMax - index);
        const placeholder = { value: limitMax - index, count: 0, total: 0 };

        return element || placeholder;
      })
      .reduce(
        (acc, item, index) =>
          acc.concat({
            ...item,
            total: index === 0 ? item.count : acc[index - 1].total + item.count,
          }),
        []
      );

    const items = range.map((item, index) =>
      this.buildItem({
        lowerBound: item.value,
        count: item.total,
        isLastSelectableItem: range.length - 1 === index,
        max: limitMax,
        translate,
        createURL,
      })
    );

    return <div {...cx('root', !canRefine && 'noRefinement')}>{items}</div>;
  }
}

export default translatable({
  ratingLabel: ' & Up',
})(StarRating);
