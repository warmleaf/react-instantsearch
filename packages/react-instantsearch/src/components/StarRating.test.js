import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import StarRating from './StarRating';

Enzyme.configure({ adapter: new Adapter() });

describe('StarRating', () => {
  it('supports passing max/min values', () => {
    const tree = renderer
      .create(
        <StarRating
          createURL={() => '#'}
          refine={() => null}
          min={1}
          max={5}
          currentRefinement={{ min: 1, max: 5 }}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('supports passing max/min values smaller than count max & min', () => {
    const tree = renderer
      .create(
        <StarRating
          createURL={() => '#'}
          refine={() => null}
          min={2}
          max={4}
          currentRefinement={{}}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies translations', () => {
    const tree = renderer
      .create(
        <StarRating
          createURL={() => '#'}
          refine={() => null}
          translations={{
            ratingLabel: ' & Up',
          }}
          min={1}
          max={5}
          currentRefinement={{ min: 1, max: 5 }}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('expect to not throw when only min is defined', () => {
    expect(() => {
      renderer.create(
        <StarRating
          createURL={() => '#'}
          refine={() => null}
          translations={{
            ratingLabel: ' & Up',
          }}
          min={3}
          currentRefinement={{ min: 1, max: 5 }}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      );
    }).not.toThrow();
  });

  it('expect to not throw when only max is defined', () => {
    expect(() => {
      renderer.create(
        <StarRating
          createURL={() => '#'}
          refine={() => null}
          translations={{
            ratingLabel: ' & Up',
          }}
          max={3}
          currentRefinement={{ min: 1, max: 5 }}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      );
    }).not.toThrow();
  });

  it('expect to render from from 0 when min is negative', () => {
    const tree = renderer
      .create(
        <StarRating
          createURL={() => '#'}
          refine={() => null}
          translations={{
            ratingLabel: ' & Up',
          }}
          min={-5}
          max={5}
          currentRefinement={{ min: 1, max: 5 }}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('expect to render nothing when min is higher than max', () => {
    const tree = renderer
      .create(
        <StarRating
          createURL={() => '#'}
          refine={() => null}
          translations={{
            ratingLabel: ' & Up',
          }}
          min={5}
          max={3}
          currentRefinement={{ min: 1, max: 5 }}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const refine = jest.fn();
  const createURL = jest.fn();
  const starRating = (
    <StarRating
      createURL={createURL}
      refine={refine}
      min={1}
      max={5}
      currentRefinement={{ min: 1, max: 5 }}
      count={[
        { value: '1', count: 4 },
        { value: '2', count: 2 },
        { value: '3', count: 3 },
        { value: '4', count: 3 },
        { value: '5', count: 0 },
      ]}
      canRefine={true}
    />
  );

  beforeEach(() => {
    refine.mockClear();
    createURL.mockClear();
  });

  it('should create an URL for each row except for the largest: the default selected one', () => {
    const wrapper = mount(starRating);

    expect(createURL.mock.calls).toHaveLength(4);
    expect(createURL.mock.calls[0][0]).toEqual({ min: 5, max: 5 });
    expect(createURL.mock.calls[1][0]).toEqual({ min: 4, max: 5 });
    expect(createURL.mock.calls[2][0]).toEqual({ min: 3, max: 5 });
    expect(createURL.mock.calls[3][0]).toEqual({ min: 2, max: 5 });

    wrapper.unmount();
  });

  it('refines its value on change', () => {
    const wrapper = mount(starRating);
    const links = wrapper.find('.ais-StarRating__ratingLink');
    expect(links).toHaveLength(5);

    let selectedLink = wrapper.find('.ais-StarRating__ratingLinkSelected');
    expect(selectedLink).toHaveLength(1);

    links.first().simulate('click');

    expect(refine.mock.calls).toHaveLength(1);
    expect(refine.mock.calls[0][0]).toEqual({ min: 5, max: 5 });

    selectedLink = wrapper.find('.ais-StarRating__ratingLinkSelected');
    expect(selectedLink).toBeDefined();

    refine.mockClear();

    const disabledLink = wrapper
      .find('.ais-StarRating__ratingLinkDisabled')
      .find('.ais-StarRating__ratingIcon');

    expect(disabledLink).toHaveLength(5);
    wrapper.unmount();
  });

  it('should display the right number of stars', () => {
    const wrapper = mount(starRating);
    wrapper
      .find('.ais-StarRating__ratingLink')
      .last()
      .simulate('click');

    const selectedLink = wrapper.find('.ais-StarRating__ratingLinkSelected');

    const fullIcon = selectedLink.find('.ais-StarRating__ratingIcon');
    const emptyIcon = selectedLink
      .first()
      .find('.ais-StarRating__ratingIconEmpty');

    expect(fullIcon).toHaveLength(1);
    expect(emptyIcon).toHaveLength(4);
    wrapper.unmount();
  });

  it('clicking on the selected refinement should select the largest range', () => {
    const wrapper = mount(starRating);
    wrapper.setProps({ currentRefinement: { min: 5, max: 5 } });

    const links = wrapper.find('.ais-StarRating__ratingLink');
    links.first().simulate('click');

    expect(refine.mock.calls).toHaveLength(1);
    expect(refine.mock.calls[0][0]).toEqual({ min: 1, max: 5 });
    wrapper.unmount();
  });

  describe('Panel compatibility', () => {
    it('Should indicate when no more refinement', () => {
      const canRefine = jest.fn();
      const wrapper = mount(starRating, {
        context: { canRefine },
        childContextTypes: { canRefine: PropTypes.func },
      });

      expect(canRefine.mock.calls).toHaveLength(1);
      expect(canRefine.mock.calls[0][0]).toEqual(true);
      expect(wrapper.find('.ais-StarRating__noRefinement')).toHaveLength(0);

      wrapper.setProps({ canRefine: false });

      expect(canRefine.mock.calls).toHaveLength(2);
      expect(canRefine.mock.calls[1][0]).toEqual(false);
      expect(wrapper.find('.ais-StarRating__noRefinement')).toHaveLength(1);
    });
  });
});
