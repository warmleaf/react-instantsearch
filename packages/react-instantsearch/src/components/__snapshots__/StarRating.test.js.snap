// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating applies translations 1`] = `
<div
  className="ais-StarRating__root"
>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      5
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      9
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      12
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      14
    </span>
  </a>
  <div
    className="ais-StarRating__ratingLink ais-StarRating__ratingLinkSelected"
    disabled={false}
  >
    <span
      className="ais-StarRating__ratingIcon ais-StarRating__ratingIconSelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingLabel ais-StarRating__ratingLabelSelected"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount ais-StarRating__ratingCountSelected"
    >
      15
    </span>
  </div>
</div>
`;

exports[`StarRating expect to render from from 0 when min is negative 1`] = `
<div
  className="ais-StarRating__root"
>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      5
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      9
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      12
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      14
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink ais-StarRating__ratingLinkSelected"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon ais-StarRating__ratingIconSelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingLabel ais-StarRating__ratingLabelSelected"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount ais-StarRating__ratingCountSelected"
    >
      15
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      15
    </span>
  </a>
</div>
`;

exports[`StarRating expect to render nothing when min is higher than max 1`] = `
<div
  className="ais-StarRating__root"
/>
`;

exports[`StarRating supports passing max/min values 1`] = `
<div
  className="ais-StarRating__root"
>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      5
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      9
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      12
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      14
    </span>
  </a>
  <div
    className="ais-StarRating__ratingLink ais-StarRating__ratingLinkSelected"
    disabled={false}
  >
    <span
      className="ais-StarRating__ratingIcon ais-StarRating__ratingIconSelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingIconEmpty ais-StarRating__ratingIconEmptySelected"
    />
    <span
      className="ais-StarRating__ratingLabel ais-StarRating__ratingLabelSelected"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount ais-StarRating__ratingCountSelected"
    >
      15
    </span>
  </div>
</div>
`;

exports[`StarRating supports passing max/min values smaller than count max & min 1`] = `
<div
  className="ais-StarRating__root"
>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      4
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      7
    </span>
  </a>
  <a
    className="ais-StarRating__ratingLink"
    disabled={false}
    href="#"
    onClick={[Function]}
  >
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIcon"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingIconEmpty"
    />
    <span
      className="ais-StarRating__ratingLabel"
    >
       & Up
    </span>
    <span>
       
    </span>
    <span
      className="ais-StarRating__ratingCount"
    >
      9
    </span>
  </a>
</div>
`;
